### PR TITLE
Payload testing UI

### PR DESCRIPTION
**Pull request by bbguimaraes**
[preview](https://payload-testing-ui-bbguimaraes.apps.ci.l2s4.p1.openshiftapps.com/runs)

https://issues.redhat.com/browse/DPTP-2542

### CODE REVIEW

File: cmd/payload-testing-ui/main.go [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+	}
+	return nil
+}
+
+const (
+	appCIContextName = string(api.ClusterAPPCI)
+)
+
+var (
+	metrics = prowmetrics.NewMetrics("configresolver")
```
Comment (petr-muller) [inline cmd/payload-testing-ui/main.go:59]: not configresolver ;)

File: cmd/payload-testing-ui/main.go [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+)
+
+func l(fragment string, children ...simplifypath.Node) simplifypath.Node {
+	return simplifypath.L(fragment, children...)
+}
+
+func main() {
+	logrusutil.ComponentInit()
+	logger := logrus.WithField("plugin", "payload-testing")
```
Comment (petr-muller) [inline cmd/payload-testing-ui/main.go:68]: Probably not a plugin?

File: cmd/payload-testing-ui/main.go [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+		if !hasInClusterConfig {
+			logrus.WithError(err).Fatalf("had no context for '%s' and loading InClusterConfig failed", appCIContextName)
+		}
+		logrus.Infof("use InClusterConfig for %s", appCIContextName)
+		kubeconfigs[appCIContextName] = inClusterConfig
+	}
+	kubeConfig := kubeconfigs[appCIContextName]
+	kubeClient, err := ctrlruntimeclient.New(&kubeConfig, ctrlruntimeclient.Options{})
+	if err != nil {
+		logrus.WithError(err).Fatal("could not get Hive client for Hive kube config")
```
Comment (petr-muller) [inline cmd/payload-testing-ui/main.go:96]: Not hive ;)

**Review by petr-muller — COMMENTED** [review]
Mostly LGTM up to several copy-paste artifacts. I think it would be reasonable to merge something like this soon-ish and implement in increments, WDYT?

**Comment by bbguimaraes** [comment]
Ready for review.  Changes from the previous version:

- I reverted to a simpler dump of `spec`/`status` for now until the final format is determined and the other components are populating them
- `main.go` is now a proper file

Ready for deployment in: https://github.com/openshift/release/pull/23358.

**Comment by bbguimaraes** [comment]
Job mapping re-re-verted, now conforming to the internal discussion.

File: pkg/api/pullrequestpayloadqualification/v1/types.go [hunk 4]
```diff
@@ -123,3 +124,9 @@ type PullRequestPayloadQualificationRunList struct {
 
 	Items []PullRequestPayloadQualificationRun `json:"items"`
 }
+
+// JobName maps the name in the spec to the corresponding Prow job name.
+// It matches the `ReleaseJobName` value in the status.
+func (s ReleaseJobSpec) JobName() string {
+	return s.CIOperatorConfig.SimpleJobName(jobconfig.PeriodicPrefix, s.Test)
```
Comment (petr-muller) [inline pkg/api/pullrequestpayloadqualification/v1/types.go:131]: Don't we need `JobName`?

(why do we enven have `SimpleJobName` / `JobName`?

Comment (bbguimaraes) [reply pkg/api/pullrequestpayloadqualification/v1/types.go]: You're right, that was based on a previous (erroneous) idea about what the job names were, I didn't notice when I reinstated it.  Corrected.

(I also found the number of random combinations of `Metadata -> string` functions amusing)

**Review by petr-muller — APPROVED** [review]

**Pull request merged**